plugins/restful/RestfulEntityBase.php

```diff
@@ -632,6 +632,12 @@ public function createEntity() {
     $bundle_key = $entity_info['entity keys']['bundle'];
     $values = $bundle_key ? array($bundle_key => $this->bundle) : array();
 
+    // Set the uid as early as possible to avoid false-negatives on permission
+    // checks.
+    if (!empty($this->getAccount()->uid) && $this->getAccount()->uid !== 0) {
+      $values['uid'] += $this->getAccount()->uid;
```

Comment:
Given that, the fix for this might be to restructure the way entities are created. Perhaps we have to build the whole entity in memory before checking permissions on it. That seems pretty messy given field-level permissions, though.